include/cpp2util.h

```diff
@@ -2281,8 +2281,15 @@ struct range
         : first{ f }
         , last{ l }
     {
-        if (include_last) {
-            ++last;
+        if (include_last) { 
+            if constexpr (std::is_arithmetic_v<T>) {
+                if (last == std::numeric_limits<T>::max()) {
```

Comment:
Now that I'm using `TT` as the possibly-widened underlying type, I think just making it `<TT>::max()` seems right, what do you think?

Reply:
Yup